shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -47,14 +89,41 @@ @interface FlutterTextInputPlugin () <NSTextInputClient>
  */
 @property(nonatomic, weak) FlutterViewController* flutterViewController;
 
+/**
+ * The affinity for the current cursor position.
+ */
+@property FlutterTextAffinity textAffinity;
+
+/**
+ * ID of the text input client.
+ */
+@property(nonatomic, readonly, nonnull) NSNumber* clientID;
+
+/**
+ * Keyboard type of the client. See available options:
+ * https://api.flutter.dev/flutter/services/TextInputType-class.html
+ */
+@property(nonatomic, readonly, nonnull) NSString* inputType;
+
+/**
+ * An action requested by the user on the input client. See available options:
+ * https://api.flutter.dev/flutter/services/TextInputAction-class.html
+ */
+@property(nonatomic, readonly, nonnull) NSString* inputAction;
```

Comment:
It's weird to declare these all as readonly when you are in fact writing to them, just not using property syntax.

Reply:
Yikes. Copy-paste context fail. Thanks!